src/commands/createIntegrationFromApim.ts

```diff
@@ -0,0 +1,140 @@
+        title: UiStrings.CreatingIntegration,
+    }, async (progress, token) => {
+        const updatedApiCenter = await enableSystemAssignedManagedIdentity(apiCenterService);
+
+        await assignManagedIdentityReaderRole(node!.subscription, updatedApiCenter, apimResourceId);
+
+        const apiSource = await createIntegration(apiCenterService, linkName, apimResourceId);
+
+        vscode.window.showInformationMessage(vscode.l10n.t(UiStrings.IntegrationCreated, apiSource.name));
+        node!.refresh(context);
+    });
+}
+
+async function selectApim(context: IActionContext): Promise<string> {
+    const subscriptionTreeItem = await ext.treeDataProvider.showTreeItemPicker<SubscriptionTreeItem>(SubscriptionTreeItem.contextValue, context);
+    const resourceGraphService = new ResourceGraphService(subscriptionTreeItem.subscription);
+
+    const apims = await resourceGraphService.listApims();
+    if (!apims || apims.length === 0) {
+        throw new Error(UiStrings.NoAPIManagementFound);
+    }
+
+    const apimNames = apims.map(apim => apim.name);
+    const selectedApimName = await vscode.window.showQuickPick(apimNames, { title: UiStrings.SelectAPIManagementService, ignoreFocusOut: true });
+    if (!selectedApimName) {
+        throw new UserCancelledError();
+    }
+
+    const selectedApim = apims.find(apim => apim.name === selectedApimName);
+    return selectedApim.id;
+}
+
+async function enableSystemAssignedManagedIdentity(apiCenterService: ApiCenterService): Promise<ApiCenter> {
+    const apiCenter = await apiCenterService.getApiCenter();
+    if (apiCenter.identity?.principalId) {
+        return apiCenter;
+    } else {
+        const apiCenterPayload: ApiCenterPayload = {
+            location: apiCenter.location,
+            identity: {
+                type: "SystemAssigned"
+            },
+            sku: {
+                name: apiCenter.sku.name
+            }
+        };
+        const updatedApiCenter = await apiCenterService.createOrUpdateApiCenterService(apiCenterPayload);
+        validateResponse(updatedApiCenter);
+        return updatedApiCenter;
+    }
+}
+
+async function assignManagedIdentityReaderRole(subscriptionContext: ISubscriptionContext, apiCenter: ApiCenter, apimResourceId: string): Promise<void> {
+    const azureService = new AzureService(subscriptionContext);
+    const roleAssignmentPayload: RoleAssignmentPayload = {
+        properties: {
+            roleDefinitionId: `/subscriptions/${subscriptionContext.subscriptionId}/providers/Microsoft.Authorization/roleDefinitions/71522526-b88f-4d52-b57f-d31fc3546d0d`, // 'API Management Service Reader' role
+            principalId: apiCenter.identity!.principalId!,
+            principalType: "ServicePrincipal"
+        }
+    };
+    const scope = apimResourceId.substring(1);
+    const response = await azureService.createOrUpdateRoleAssignment(scope, roleAssignmentPayload);
+    if (![200, 201, 409].includes(response.status)) {
```

Comment:
no 409 for such apis in the doc
https://learn.microsoft.com/en-us/rest/api/authorization/role-assignments/create?view=rest-authorization-2022-04-01&tabs=HTTP#response